sdk/inc/azure/core/az_json.h

```diff
@@ -420,6 +427,13 @@ AZ_NODISCARD az_result az_json_writer_append_string(az_json_writer* ref_json_wri
  * @remarks The function validates that the provided JSON to be appended is valid and properly
  * escaped, and fails otherwise.
  *
+ * @note If you receive an #AZ_ERROR_NOT_ENOUGH_SPACE result while appending data for which there is
```

Comment:
We aren't consistent on whether note should go right after `@brief` or after `@param`.

Which made me realize, we aren't consistent on order of `@remarks` and `@return` either, which might be a good thing to fix across the board.